routing/routing.go

```diff
@@ -154,19 +161,73 @@ func New(o Options) *Routing {
 
 	r := &Routing{log: o.Log, quit: make(chan struct{})}
 	initialMatcher, _ := newMatcher(nil, MatchingOptionsNone)
-	r.matcher.Store(initialMatcher)
+	rt := &routeTable{
+		m:       initialMatcher,
+		created: time.Now().UTC(),
+	}
+	r.routeTable.Store(rt)
 	r.startReceivingUpdates(o)
 	return r
 }
 
+// ServeHTTP renders the list of current routes.
+func (r *Routing) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	rt := r.routeTable.Load().(*routeTable)
+	req.ParseForm()
+	createdUnix := strconv.FormatInt(rt.created.Unix(), 10)
+	ts := req.Form.Get("timestamp")
+	if ts != "" {
+		if createdUnix != ts {
+			http.Error(w, "invalid timestamp", http.StatusBadRequest)
+			return
+		}
+	}
+	offset, err := extractParam(req, "offset", 0)
+	if err != nil {
+		http.Error(w, "invalid offset", http.StatusBadRequest)
+		return
+	}
+	limit, err := extractParam(req, "limit", defaultRouteListingLimit)
+	if err != nil {
+		http.Error(w, "invalid limit", http.StatusBadRequest)
+		return
+	}
+	accept := req.Header.Get("Accept")
+	routes := slice(rt.validRoutes, offset, limit)
+
+	// TODO: Maybe this is too strict?
+	if accept == "application/json" {
+		w.Header().Set(routeTableTimestampHeaderName, createdUnix)
+		w.Header().Set("content-type", "application/json")
+		if err := json.NewEncoder(w).Encode(routes); err != nil {
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// TODO: Maybe this is too strict? Also, what is the correct MIME type for eskip?
+	if accept == "application/eskip" {
+		w.Header().Set(routeTableTimestampHeaderName, createdUnix)
+		w.Header().Set("content-type", "application/eskip")
+		eskipRoutes := make([]*eskip.Route, len(routes))
+		for i, r := range routes {
+			eskipRoutes[i] = r
+		}
+		fmt.Fprint(w, eskip.String(eskipRoutes...))
```

Comment:
This sounds like a good idea. I'd even go further and see how `ls` understands that it's output is being piped to another process and change the behavior so that chaining multiple commands work "well" in the unix world